spec/rubocop/cop/style/disable_cop_for_entire_file_in_source_code_spec.rb

```diff
@@ -0,0 +1,134 @@
+      false
+      # rubocop:enable Department/CopName
+    RUBY
+  end
+
+  it 'registers an offense for block directives wrapping the only line of code, even if it has an inline directive' do
+    expect_offense(<<~RUBY)
+      # rubocop:disable Department/CopName
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ When disabling cops for an entire file, prefer excluding the file in `.rubocop.yml`.
+      true # rubocop:disable Department/AnotherCopName
+      # rubocop:enable Department/CopName
+    RUBY
+  end
+
+  it 'registers an offense for nested block directives wrapping all the code' do
+    expect_offense(<<~RUBY)
+      # rubocop:disable Department/CopName
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ When disabling cops for an entire file, prefer excluding the file in `.rubocop.yml`.
+      # rubocop:disable Department/AnotherCopName
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ When disabling cops for an entire file, prefer excluding the file in `.rubocop.yml`.
+      true
+      # rubocop:enable Department/AnotherCopName
+      # rubocop:enable Department/CopName
+    RUBY
+  end
+
+  it 'registers only offenses for block directives wrapping all the code, not all block directives' do
+    expect_offense(<<~RUBY)
+      # rubocop:disable Department/CopName
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ When disabling cops for an entire file, prefer excluding the file in `.rubocop.yml`.
+      123
+      # rubocop:disable Department/AnotherCopName
+      456
+      # rubocop:enable Department/AnotherCopName
+      # rubocop:enable Department/CopName
+    RUBY
+  end
+
+  it 'registers no offenses when there is code after the enable directive' do
+    expect_no_offenses(<<~RUBY)
+      # rubocop:disable Department/CopName
+      false
+      # rubocop:enable Department/CopName
+      true
+    RUBY
+  end
+
+  it 'registers no offenses when there is code before the disable directive' do
+    expect_no_offenses(<<~RUBY)
+      true
+      # rubocop:disable Department/CopName
+      false
+      # rubocop:enable Department/CopName
+    RUBY
+  end
+
+  it 'registers no offenses when the file is empty except for comments' do
+    expect_no_offenses(<<~RUBY)
+      # rubocop:disable Department/CopName
+      # rubocop:enable Department/CopName
+    RUBY
+  end
+
+  it 'registers no offenses when the file is empty' do
+    expect_no_offenses('')
+  end
+
+  it 'registers no offenses for single inline directive' do
+    expect_no_offenses(<<~RUBY)
+      true # rubocop:disable Department/CopName
+    RUBY
+  end
+
+  it 'registers no offenses when disabling all cops' do
+    # Because `rubocop:disable all` disables all cops, it disables this one too.
+    # Ideally, it would probably still apply, but then this cop would need to be a special case.
+    expect_no_offenses(<<~RUBY)
+      # rubocop:disable all
+      false
+      # rubocop:enable all
+    RUBY
+  end
```

Comment:
I think changing this would require either making this cop a special case, or changing block directives to not apply to the lines that control them, which would necessitate a change to `rubocop-ast`. I'm not sure of the implication of making such a change.

Reply:
This is an interesting edge case. I think we should just leave it because if you want to disable all cops, that shouldn't have special cased cops that are not included.